pkg/util/helper/binding.go

```diff
@@ -317,21 +317,21 @@ func GetResourceBindings(c client.Client, ls labels.Set) (*workv1alpha2.Resource
 	return bindings, c.List(context.TODO(), bindings, listOpt)
 }
 
-// DeleteWorkByRBNamespaceAndName will delete all Work objects by ResourceBinding namespace and name.
-func DeleteWorkByRBNamespaceAndName(c client.Client, namespace, name string) error {
-	return DeleteWorks(c, namespace, name)
+// DeleteWorkByResourceBinding will delete all Work objects by ResourceBinding namespace and name.
+func DeleteWorkByResourceBinding(c client.Client, bindingMeta *metav1.ObjectMeta) error {
+	return DeleteWorks(c, bindingMeta)
 }
 
-// DeleteWorkByCRBName will delete all Work objects by ClusterResourceBinding name.
-func DeleteWorkByCRBName(c client.Client, name string) error {
-	return DeleteWorks(c, "", name)
+// DeleteWorkByClusterResourceBinding will delete all Work objects by ClusterResourceBinding name.
+func DeleteWorkByClusterResourceBinding(c client.Client, bindingMeta *metav1.ObjectMeta) error {
+	return DeleteWorks(c, bindingMeta)
 }
 
 // DeleteWorks will delete all Work objects by labels.
-func DeleteWorks(c client.Client, namespace, name string) error {
-	workList, err := GetWorksByBindingNamespaceName(c, namespace, name)
+func DeleteWorks(c client.Client, bindingMeta *metav1.ObjectMeta) error {
+	workList, err := GetWorksByBinding(c, bindingMeta)
 	if err != nil {
-		klog.Errorf("Failed to get works by ResourceBinding(%s/%s) : %v", namespace, name, err)
+		klog.Errorf("Failed to get works by ResourceBinding(%s/%s) : %v", bindingMeta.Namespace, bindingMeta.Name, err)
```

Comment:
It may not be appropriate to use ResourceBinding here, as it could be ClusterResourceBinding. Can you help modify it.